common/messages/messageTrace.go

```diff
@@ -39,38 +39,49 @@ func checkFileName(name string) bool {
 	if globals.Params.DebugLogRegEx == "" {
 		return false
 	}
+	checkForChangesInDebugRegex()
+	flag, old := enabled[name]
+	if !old {
+		flag = TestRegex.Match([]byte(name))
+		enabled[name] = flag
+	}
+	return flag
+}
+
+func checkForChangesInDebugRegex() {
 	// if  the regex string has changed ...
 	if globals.Params.DebugLogRegEx != globals.LastDebugLogRegEx {
+		lastSlashIndex := strings.LastIndex(globals.Params.DebugLogRegEx, string(os.PathSeparator))
```

Comment:
Can you refactor this logic of separating a string into the location and regex into a function? 

I see your unit test going as far as launching a simtest, but it would be nice to have just this functionality unit tested against various strings. It would also then consistently run in circleci.